modules/programs/nheko.nix

```diff
@@ -0,0 +1,87 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.nheko;
+
+  iniFmt = pkgs.formats.ini { };
+
+  configDir = if pkgs.stdenv.hostPlatform.isDarwin then
+    "Library/Application Support"
+  else
+    config.xdg.configHome;
+
+  camelCaseToSnakeCase =
+    replaceStrings upperChars (map (s: "_${s}") lowerChars);
```

Comment:
We typically don't bother doing this for RFC-42 type `settings` options. That is, having 

    general.disable_certificate_validation = false

is OK.

Reply:
Can we have it this way then? Or RFC-42 prohibits this? I really dislike snake case on nix settings by default 😅

Reply:
Typically, the description of `settings` options contain an upstream link to where the user can find information about available program options. That upsteam documentation would show the snake-case names so it may confuse a user when they see non-snake-case names here.

But, I have no strong feeling about it so you can keep the `camelCaseToSnakeCase` if you want. As long as it is unambiguous, i.e., nheko doesn't have any option that contains a uppercase character.

Reply:
I'm afraid all these settings are very much under-documented, I had to dig them up, so no documentation link...